### PR TITLE
Cacheless ver1 4-2-stable

### PR DESCRIPTION
**Pull request by JustinKyleJames**
[1849] Add support for cacheless s3.

### CODE REVIEW

**Comment by trel** [comment]
So, all of these different issues are covered by this pull request?

And the ones that don't have an issue number, can/should they be squashed into ones that do?

**Comment by JustinKyleJames** [comment]
This is not passing test in Centos due to thread_local issue.  Do not merge until this is fixed.

**Comment by JustinKyleJames** [comment]
I have fixed all of the codacy issues I can.  This can be merged.

**Comment by trel** [comment]
If tests are all passing, both for cached and cacheless...
let's add # everywhere...

i notice one commit with different email address, and please make sure every commit references an issue (or gets squashed).

**Comment by JustinKyleJames** [comment]
Will do.


________________________________

File: packaging/resource_suite_s3_nocache.py [hunk 1]
```diff
@@ -1373,7 +1373,7 @@ def test_detached_mode(self):
         file1 = "f1"
         file2 = "f2"
         #resource_host = test.settings.HOSTNAME_3
-        resource_host = "irods.org"
+        resource_host = "152.54.3.140"  # irods.org
```
Comment (trel) [inline packaging/resource_suite_s3_nocache.py]: need to remove the specific IP

File: packaging/resource_suite_s3_nocache.py [hunk 2]
```diff
@@ -1413,7 +1413,7 @@ def test_attached_mode_invalid_resource_host(self):
 
         file1 = "f1"
         #resource_host = test.settings.HOSTNAME_3
-        resource_host = "irods.org"
+        resource_host = "152.54.3.140"  # irods.org
```
Comment (trel) [inline packaging/resource_suite_s3_nocache.py]: need to remove the specific IP

**Comment by JustinKyleJames** [comment]
I made the changes and re-ran those tests.  It should be ready to merge.

**Pull request merged**